### DISCUSSION TITLE
AI — we want your input (emerging patterns, use cases, research findings)

### DISCUSSION THREAD

**Discussion by kubabartwicki** [alphagov/govuk-design-system#3804] [AI]
Hello

We are starting a discussion about emerging design patterns for utilising AI in public services. This probably mostly covers generative AI, but other design patterns for making use of machine learning, data science, and similar technologies are also useful.

We’re doing this because there is a lot here! Many teams across government are experimenting with AI, and we are working out how to design for this. We need to work out how to surface AI-generated content to users, how to talk to them about accuracy, and how to design more dynamic interactions. A space to share learnings in the open seems like a good place to start 🥁

I work on the GOV.UK AI team, where we are experimenting how AI can help people navigate the complexity of government services and information. We have a ton of insights from testing so far, and we know that other teams do too. Most of our learnings come from experimenting with GOV.UK Chat. You can learn more about that in [this blog post](https://insidegovuk.blog.gov.uk/2024/01/18/the-findings-of-our-first-generative-ai-experiment-gov-uk-chat/), or drop us a message if you want to chat more about it.

We’ll start by sharing some of our insights, but please join the discussion if you work in government and have things to share. Keep in mind that it’s all work in progress, and evolving fast!

**Comment by LucyMcArthur-GDS** [comment]
### Describing accuracy limitations in an AI chat

Generative AI can sometimes show inaccurate information with a high degree of authority. We’ve been learning lots about how to communicate this and make sure users don’t follow inaccurate advice.

In the GOV.UK Chat experiment, we’ve had some success explaining accuracy limitations in a chat conversation, as part of the onboarding. Showing onboarding as messages received from the chat was a lot more effective than displaying the content on a static page. It also allows us to break the information down into shorter messages without overwhelming users.

We are also experimenting with integrating actions into the chat interface. During the onboarding, the bottom area of the chat shows two buttons. Once you’ve onboarded, this changes to include a text input field. Users responded really well to having rich UI elements in the chat, and it’s a pattern we want to build on. We have some really interesting content insights on this which we’ll share soon too.

<img width="375" alt="Screenshot of a chat interface including a message and confirmation buttons" src="https://github.com/alphagov/govuk-design-system/assets/168846230/d88dec63-c811-4fb5-aadc-59bc09a2743d">

**Reply by Becca-Ayre** [reply]
This is great to see. The tool I am supporting has these caveats on a static page, so it is great to see another approach.

Encouraging that you got good interaction from users with this approach. How did this convert later in the journey? Can you see whether users are heeding the advice and checking the answers?

**Comment by LucyMcArthur-GDS** [comment]
### Checking answers

We also need to give users a really clear next step to do what they’re trying to do, and to make it easy for them to verify information generated by AI. 

In our chat experiments, we are integrating a “check this answer” step in the user journey. This includes links to relevant GOV.UK guidance pages directly after chat’s responses. In research, we found that links were viewed as a way to find further information, but not always as a checking step, so we made this more prominent. Checking content isn’t a typical user behaviour within digital tools, but using a prominent “check this answer” component with a warning icon helped encourage it.

To make this more intuitive, we are also exploring things like linking people to specific parts of pages, or highlighting relevant content snippets.

<img width="265" alt="Screenshot of chat interface showing chat's answer followed by links to relevant web pages" src="https://github.com/alphagov/govuk-design-system/assets/168846230/8269a0e8-f450-4fda-b305-497e598b7b9c">

**Reply by munhitsu** [reply]
How about referencing specific paragraph, sentence? I know it is harder to implement but it also makes reference checking faster.

**Reply by LucyMcArthur-GDS** [reply]
Thanks for your comment, it would be cool to explore referencing within the answer which I think is what you are suggesting (e.g. how Perplexity have numbers after specific sentences in the answers). We also explored highlighting specific sentences on a linked page, which is useful for users if their question is less complex.

**Reply by Becca-Ayre** [reply]
Hello! I am working with a team in DWP who are developing a Q&A response tool. Great to see you are looking at similar design problems! 

We got some feedback from our private beta users that they wanted to see the specific paragraph or sentences highlighted, and I was told by data science and engineering colleagues that its technically possible but he processing power required to do it would be quite high.

Have you got any evidence yet of whether users **are** checking answers using these references? And if they are, how confident they are in doing the checks? We are still to really test this with our tool but I have raised this as a risk we need to explore further.

**Comment by uxCharlotteR** [comment]
Thanks for starting this thread :)
I'm working on patterns and components for gen AI as part of the work we're doing in the Incubator for AI (i.AI) , here in Cabinet Office.
I've started some conversations with Tim Paul and Charlotte Downs, in GDS, about how we might collaborate across gov to experiment and iterate on emerging AI components. But it'd be great to be more connected with others about what is happening across gov departments to all be helping each other with insights and learnings.

I've also been looking at other gen AI components outside of gov that we might be able to learn and experiment, here in the Incubator.
Lots of exciting times ahead. 👍

**Reply by kubabartwicki** [reply]
Nice, Tim mentioned this, would be good to chat!

**Comment by EvangelineWright** [comment]
Hello! I have just come across this thread and would love to connect. I am currently working on designing and developing AI powered services within the DWP. We have been using the GDS design system but have needed to deviate due to AI design requirements, such as transparency and user co-creation of content. It would be brilliant to discuss these and see what other organisations are doing across gov!

**Reply by kubabartwicki** [reply]
Oh thanks for reaching out Evangeline! Would you be able to share some approaches on this thread? If not, happy to set up a chat if you email me at kuba.bartwicki@digital.cabinet-office.gov.uk

**Reply by Becca-Ayre** [reply]
Hello. Did Evie ever get back to you Kuba? I have picked this work up from Evie since earlier this year.

**Comment by nicolaatd** [comment]
I'd be really interested to see what consideration is being given to the environmental impact of the use of AI in cases like this. I think there are definitely questions to be asked on the value of generative AI vs the sustainability of it.

**Comment by AntoniaRub** [comment]
I've just come across this via the AI conference being run at the moment. I am currently looking at the Gen AI use in the JAC but also considering the inclusivity/diversity aspect of Gen AI and what we need to consider, taking into account how fast moving this topic is at the moment to ensure that this remains something we take into account.

**Comment by uxCharlotteR** [comment]
There's a cross-Gov Slack channel for folk doing UCD in AI products. [#ai-ucd](https://ukgovernmentdigital.slack.com/archives/C08BLLQS0JH)
Feels like there are lots of us wanting to share together in this space. I know there is this group but thought I'd share it too. Let me know if anyone wants to be added.

**Comment by romanidavies** [comment]
We have been working on a white label version of the AI assistant created for caseworker guidance.

We have replaced the content of our designs with a general subject matter. This subject matter is 'Superhero applications', however we hope this will still illustrate the design decisions made.

It can be found here: https://github.com/LandRegistry/ai-assistant-whitelabel 

For context: 

Within HMLR, Andy Porter, Kit Harding and I have been developing a ‘design language’ for AI UI.

There has been very little validation or usability testing or feedback from users at this time. We appreciate this is a completely new paradigm, and we don’t have gov patterns yet so wanted to share our progress.

We’ve been working on: 
- General look and feel like layout, colours
- What language to use- what to name things, how to describe things- including a how to ask/ use AI
- New components/patterns

Within these files we have included some insight into the decisions made. They are duplicated here: 

### Guidance homepage: 

We needed to consider how the assistant could be integrated with the existing caseworker guidance system.

<img width="1420" height="995" alt="Homepage white label" src="https://github.com/user-attachments/assets/9b354bd8-39ea-4be6-b5f3-6459be78770b" />

Within the scope of this project, we had limited capacities to amend the existing guidance interface, therefore we had to take this into consideration of our own designs. The keyword search was already present.

We added a new lead paragraph to the page, to inform users they could search by two ways and ensured both search methods had equal weighting. It was important to present the methods equally but also try to encourage users to use the assistant.

We came up with a 'widget-like' component to represent the assistant, which could be minimised for other pages in the guidance. However, on the homepage the widget is fully expanded for maximum engagement. This widget has the same visual language as the assistant. We have included a realistic, common example question, to help users understand how to use the tool and consider how it could help their work. It was important that this example used natural language to demonstrate this benefit of interacting with the assistant.

As we are encouraging users to check the responses using the existing keyword search, the assistant opens in a new tab.

### Caseworker guidance article page

Following a keyword search on the homepage, users are taken to a search results page where they can then select the appropriate article from the shortlisted view.

<img width="1420" height="995" alt="Article page white label" src="https://github.com/user-attachments/assets/b8010910-afdf-4176-920c-84040b10af5d" />

We believe that if users are unable to easily find the information using the traditional, keyword search, they may consider utilising the tool. Therefore, users can launch the assistant from this article page.

On this page, we have added the minimised widget to this page. The widget has been minimised by removing the example text. The visual language and proactive, encouraging tone remains.

A future consideration could be that if the assistant is launched from this page, the context of the article could be passed to the model, initiating the search.

### Casework guidance AI assistant- homepage

The homepage of the assistant is minimal. We have tried to focus the user on a simple interaction- asking a question.

<img width="1288" height="701" alt="assistant homepage white label" src="https://github.com/user-attachments/assets/438e3b5e-3ffe-4593-bb4d-011c586f5be2" />

We have told users the assistant uses AI, which is important from an ethical perspective. This clause involved a lot of wider conversations, to balance the ethical responsibility of informing users of the AI usage but also that the information supplied should be checked.

To label the button we considered alternatives such as 'Ask', 'Search' and 'Ask a question'. As a conversational interface we were unsure of the language that is best suited for this component. For the first iteration, we have used 'Go'.

The LLM has been designed to allow users to have 4 question 'conversations'. This means the LLM will consider the context of up to 4 previous questions to inform the following response. Up to 4 questions was considered the optimal number of questions, that best informed the LLM without degrading the quality of responses.

The secondary link, 'Assist with another query', starts a new 'conversation' removing all previous context.

Normally, we wouldn't explain how to use a tool. However, considering this new paradigm of design, underpinned by our user research we felt there was a user need to provide guidance. Our user research found that our users didn't know how to interact with the assistant and as a result, the quality of responses were poor.

Therefore, we have included links to the purpose built guidance and demonstrations to help our users understand how to best ask questions. These materials have been tested and iterated following exposure to users.

### Casework guidance AI assistant- query

Upon entering a query into the assistant, the assistant generates a response using source information from the articles in the existing caseworker guidance system. It then considers the information found and generates a response, with the source information extracted from the articles available for users to check the AI generated response.

<img width="1288" height="1211" alt="assistant first query white label" src="https://github.com/user-attachments/assets/e0a37ec2-51b8-43da-ba7d-4d9d3741f3bf" />

The conversation is left aligned, as the assistant will only be used on desktop. This also felt more appropriate to follow the users likely reading habits on desktop (from top to bottom). We made the assumption that as the responses are likely to be large blocks of text, they would be easier to read.

Also, we have chosen to not name the questions and responses (indicating 'who was speaking') as it felt that this added an informality to the conversation, which contradicts the subject matter and use cases.

The content on the right hand panel remains consistent.

### Casework guidance AI assistant- additional queries

Our system is able to take context from up to 4 queries when generating responses. Therefore, we had to consider how multiple related queries and responses could be displayed.

<img width="1288" height="2084" alt="assistant second query white label" src="https://github.com/user-attachments/assets/042a20ce-0d46-4d43-aea8-f6814f2512cb" />

One way we distinguish between the question and response is the text widths. The users question is displayed at 1/3 width, as it is likely the response will be much longer. In addition to this, the question is on #D2E2F1, whilst the response is on govuk-colour("light-grey") #f3f2f1- these colours have been used on the GOV.UK Chat (experimental). There is a small gap between the boxes to create another visual difference between the question and response.

### Casework guidance AI assistant- inline feedback

Following the response, we have introduced inline feedback to allow the user to feed back on the helpfulness of the response.

<img width="1288" height="1211" alt="assistant first query white label" src="https://github.com/user-attachments/assets/e1071d48-b190-4886-8ad0-5f977dccf4bd" />

Initially, the team were keen to introduce a 'thumbs up'/'thumbs down' mechanism to gain feedback on the response. We were aware of the accessibility limitations that this would introduce and therefore considered current feedback approaches, specifically on the GOV.UK chat. Following this, we explored 'When you have finished, tell us if the answer was useful'. However, this approach did not allow feedback on a 'repsonse by response' basis, rather the conversation as a whole, which was too broad.

Further conversations confirmed the objective for the team was to recieve definitive feedback to the helpfulness of the response, which could then indicate further explorations into inaccurately generated responses. The 'yes'/'no' question allowed the team to be able to pinpoint inaccurate responses to improve the LLM.

<img width="764" height="138" alt="Screenshot 2025-07-31 at 09 18 34" src="https://github.com/user-attachments/assets/ef3599d3-afc9-481a-a975-e9f14fcb8190" />

We provided the team with visually hidden text for the 'yes'/'no' links to ensure they could be understood by assistive technology.

<img width="809" height="240" alt="assistant first query unhelpful white label" src="https://github.com/user-attachments/assets/b2bd3149-e900-4eb8-8a63-1ca1c0b11782" />

<img width="809" height="240" alt="assistant first query helpful white label" src="https://github.com/user-attachments/assets/98a3712b-71db-4ab9-9ee0-9a486604a0f6" />

Upon selecting 'yes'/'no' the inline text changes to ask a follow up question asking if the user would like to expand on why the response was helpful/unhelpful.

We considered which pair of words would be appropriate for a positive/negative response. This included 'useful'/'not useful', 'correct'/'incorrect', 'helpful'/'unhelpful', 'accurate'/'inaccurate', 'like'/'dislike'. We don't know which word pair works best for users as we were unable to test these but we thought 'helpful'/'unhelpful' were best as it allows users to consider their own context whereas 'correct'/'incorrect' is too absolute. For example, a response could be factually correct but does not specifically answer the users response or vice versa.

<img width="843" height="435" alt="assistant helpful feedback" src="https://github.com/user-attachments/assets/3915384f-04ef-40f9-a61d-07f1cedd82c6" />

<img width="843" height="435" alt="assistant unhelpful feedback" src="https://github.com/user-attachments/assets/39017cab-1fe0-43e8-8230-93c2de1f2a0c" />

By selecting the links following the repsonses, users are taken to one of these pages where they can provide details about why a response was helpful or unhelpful. This is optional, users do not need to add this additional detail for us to record the initial helpful/unhelpful response. The question is tailored depending on if a user answered 'yes' or 'no'.

### Casework guidance AI assistant- general feedback

As with any new tool, we are keen to facilitate users submitting feedback. This feedback is also important to improve our LLM.

<img width="1288" height="626" alt="assistant general feedback white label" src="https://github.com/user-attachments/assets/20a6fe0d-67ce-4b13-aa5f-15392fb20f01" />

To make it easier for users to submit feedback, we provide users with a free text field, rather than asking specific questions.

This is accessible through the right hand panel on the assistant's interface. Users can access this at any point during their journey. This is important as some of our users may want to feedback at a more convenient time (rather than following specific responses). We have tailored this question to be more general, to allow users to reflect on their experience using the tool as a whole. We're following the principles of the 'Tell us how to improve this service' pattern.

**Comment by anitalai-dwp** [comment]
Hello! Our team on the Jobs and Careers Service have been designing and testing an AI-powered chatbot to provide employment support for people in the UK. We've tried to stick to existing GDS patterns and some patterns I noticed from the GOV.UK chat, but have ran into a lot of limitations. For example, being able to select more than one prompt as answers, use voice input for a Mock interview tool.

It'd be good to see what other chat/AI patterns that anyone has been experimenting with, so we can learn from each other. Also super keen to get involved in creating such patterns to add to the design system.

**Comment by Becca-Ayre** [comment]
[heart]         Rebecca Ayre reacted to your message:
________________________________

**Comment by andrewmclean-hmrc** [comment]
Hi, I'm wondering if the GOV.UK chat project has explored any established industry conversation design patterns. I'm chiefly thinking about Acknowledgement, Confirmation, Prompt that's the "mantra" pattern for Hans Van Dam at the Conversation Design Institute.

The pattern is designed to build trust and confidence in users by showing them that the AI has heard them, understood them and wants to be helpful (similar to an interaction with a live call agent).

An example might be:

User: I want to register for VAT
AI: Ok, here's how to register for VAT. Gives them answer. Is there anything else I can help with?

In this example:

**Acknowledgement** = Ok (user feels they've been heard)
**Confirmation** = here's how to register for VAT (this would be an implicit confirmation - AI has over 50% confidence in its understanding of the query. Explicit confirmation would try to get the user to confirm the ask if the AI was less confident)
**Prompt** = Is there anything else I can help with? (user's turn and also way to keep them on task)

This pattern's widely used by, among others:

- Klarna
- Bank of America
- KLM
- Sephora
- Domino's

As I say, just looking to understand if this kind of pattern has been researched. I'm keen to look at it for Ask HMRC online as we move to a Contact Centre as a Service platform. So any insight you might have would be useful.

Cheers

**Comment by dodi1978** [comment]
Hi all, I just came across this forum and wanted to jump in from a slightly different angle. I am not a designer by the way, but an academic whose research interfaces with conversation design. Here is my portfolio: https://portfoliodippold.netlify.app/  
I've been researching how people experience AI chat in public services - specifically looking at rapport, i.e. that sense of warmth, trust and being understood (or not) when you're talking to a chatbot. My fieldwork has been on an NHS AI chat system used for cervical screening booking, but my findings are relevant to the kinds of questions you're raising here.
My main point perhaps: language matters! Just a few highlights:
- Choice-oriented language landed really well. Repetitive, inflexible or imperative messaging (even when well-intentioned, e.g. nudging someone to complete a step) came across as pushy and actually put people off engaging further.
- This mattered a lot more for users with prior trauma, anxiety, or who were otherwise more vulnerable - which feels relevant given how much of gov AI chat touches sensitive, high-stakes stuff (health, benefits, legal status...).
I pulled this together into a taxonomy of six "rapport dimensions" (goals, autonomy, attention, face, fairness, ethicality) that might work as a lightweight heuristic alongside other content design principles you are using.  whatever accuracy/content design principles you're developing.
I have attached a policy brief I have recently put together based on my work, which also features the rapport taxonomy. Happy to talk you though it if anyone is interested!
[Policy Brief UK GOV.pdf](https://github.com/user-attachments/files/32376158/Policy.Brief.UK.GOV.pdf)
